pkgs/desktops/gnome-3/extensions/wsmatrix.nix

```diff
@@ -0,0 +1,35 @@
+{ stdenv, fetchFromGitHub, glib }:
+
+stdenv.mkDerivation rec {
+  name = "gnome-shell-wsmatrix-${version}";
```

Comment:
Most of the gnome shell extensions use the convention `gnome-shell-extension-...` instead of `gnome-shell-...`. I don't have an opinion here other than "consistency is good", so unless others object, please add `-extension` here.